workspaces/jenkins/plugins/jenkins-backend/src/service/JenkinsBuilder.ts

```diff
@@ -153,8 +153,19 @@ export class JenkinsBuilder {
             projects: projects,
           });
         } catch (err) {
-          // Promise.any, used in the getProjects call returns an Aggregate error message with a useless error message 'AggregateError: All promises were rejected'
-          // extract useful information ourselves
+          // Get status and reason from the error handler
+          const { status, reason } = this.handleError(err);
+
+          const config = this.env.config;
+          response.status(status).json({
+            errorReason: reason,
+            connectionIssueMessage: config.getOptionalString(
```

Comment:
Is there a default value for this in case it's not set? It might be a little confusing to get back an empty string or `undefined`.

Reply:
Ah ok, yup makes sense to me, thanks

Reply:
I'm not sure about this. It might be worth hardcoding the default message in the frontend instead, as there is no real advantage to keeping it in the backend.